example_school_library_decorator/main.rb

```diff
@@ -13,66 +13,70 @@ def initialize
   end
 
   # *List all books
-  def list_books
-    @books.each do |book|
-      puts "Title: \"#{book.title}\", Author: #{book.author}"
+  def list_of_books
+    if @books.length.positive?
+      @books.each { |book| puts "title: #{book.title}, Author: #{book.author}" }
+    else
+      puts 'There are no books created so far'
     end
   end
 
   # *List all people
-  def list_people
-    @people.each do |person|
-      puts "[#{person.classroom}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+  def list_of_people
+    if @people.length.positive?
+      @people.each do |person|
+        puts "Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    else
+      puts 'There are no people created'
     end
   end
 
   def create_person
-    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
-
-    person_type = gets.chomp
-    if person_type != '1' && person_type != '2'
-      puts 'Invalid option'
-      return
-    end
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    print 'Choose either (1) or (2): '
 
-    print 'Age: '
-
-    age = gets.chomp
+    person_type_selection = gets.chomp
+    case person_type_selection
+    when '1'
+      print 'Age: '
+      age = gets.chomp.to_i
 
-    print 'Name: '
+      print 'Name: '
+      name = gets.chomp.to_s
 
-    name = gets.chomp.to_s
+      print 'Has parent permission? [Y/N]: '
+      permission = gets.chomp
+      permission = permission.downcase == 'y'
 
-    person =
-      case person_type
-      when '1'
-        print 'Has parent permission? [Y/N]: '
+      @people << Student.new(age, name, permission)
 
-        parent_permission = gets.chomp
-        parent_permission = parent_permission.downcase == 'y'
+      puts 'student created successfully'
+    when '2'
+      print 'Age: '
+      age = gets.chomp
 
-        Student.new(age, name, parent_permission)
-      when '2'
-        print 'Specialization: '
+      print 'Name: '
+      name = gets.chomp
```

Comment:
- [ ] This `create_person` method, as well as `create_rental` and `create_book` methods defies the __Single Responsibility Rule__. To follow the __Single Responsibility Rule__, a method should have one and only one responsibility